src/classes/TestFactory.cls

```diff
@@ -1,86 +1,143 @@
 @isTest
 public class TestFactory {
 
-	public static SObject createSObject(SObject sObj) {
+	// Cached metadata structures
+	public static Map<String, List<Schema.DescribeFieldResult>> requiredFieldsMap = new Map<String, List<Schema.DescribeFieldResult>>();
+	public static Map<String, Map<String, Schema.SObjectField>> sobjectToFieldsMap = new Map<String, Map<String, Schema.SObjectField>>();
+	public static Map<String, List<String>> uniqueFieldsMap = new Map<String, List<String>>();
+
+	public static SObject createSObject(SObject sObj, Boolean autoSetRequiredFields) {
```

Comment:
This is a breaking change I don't want to introduce. There is already a method with this signature that does the insert. I know you made a new method for inserting, but since people could be using the old method, I don't want the method to behave differently now.

Reply:
Fair enough... should have extended instead of changing the signature directly